### ISSUE TITLE
convexAuth handling of Access Token and Refresh Token

### ISSUE THREAD

**Issue by mez** [get-convex/convex-auth#135]
TL;DR version is asking if we can get convexAuth to support something like this: https://clerk.com/docs/references/backend/user/get-user-oauth-access-token

I am currently using convexAuth for user identity and this works fine, but in my development journey I now have to handle making API requests on behalf of the user to services providers. Would be nice if I can just handle the access token retrieval using convexAuth. We already are handling most of the leg work.

**Let's take google as an example.**

1. I configure the provider with the following
```
authorization: {
        params: {
          access_type: "offline",
          prompt: "consent",
          scope:
            "https://www.googleapis.com/auth/calendar.events https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email openid",
        },
      },
```
2. This will return in the profile callback access token and refresh token in the Tokenset
```
profile(profile, tokens) {
        const refresh_token_expires_at =
          Math.floor(Date.now() / 1000) +
          Number(tokens.refresh_token_expires_in);

        return {
          id: profile.sub,
          name: profile.name,
          email: profile.email,
          image: profile.picture,
          accessToken: tokens.access_token,
          refreshToken: tokens.refresh_token,
          // expires_at: tokens.expires_at,
        };
      },
```
3. I want to have that token info saved in the authAccounts table. Maybe the AccountCallback can be used here?
4. it'll be my responsibility to make sure refreshing is happening I am guessing. Since Auth.js doesn't handle this.

I am also looking for guidance on how to handle this concept with convex auth. Any feedback would be helpful!

**Comment by mez** [comment]
I was able to throw the token info in the user document, but as you can see, as the number of providers grow, I'll need to keep adding fields for each provider.

**Comment by mez** [comment]
I've done some thinking and I agree it's outside of this library's scope. I am indeed handling refreshing the tokens myself. I think a happy compromise would be to surface the TokenSet that comes back if the authorization is modified in the provider config. Would that be possible? The AccountCallback in the provider config currently doesn't seem to do anything.

**Comment by thomasballinger** [comment]
Yeah exposing the TokenSet seems ok unless authAccounts needs control of it in the future — I want to check, but this is likely fine, in the case that something like`[get-user-oauth-access-token](https://clerk.com/docs/references/backend/user/get-user-oauth-access-token)` were offered in the future it could be a breaking change.

Let's figure out what's wrong with `account`, how are you using that? Is that the interface you'd want here, or specifically just the token set?

**Comment by mez** [comment]
I guess nothing is wrong with the account callback. I misspoke. I just couldn't figure out how to get a hold of that information coming back when I had a convex ctx. Currently I am piggybacking the token info onto the profile record because I do get that info in `async afterUserCreatedOrUpdated(ctx: MutationCtx, args)`  callback. The cons being I needed to add extra fields to the user table that are not used after the fact.

I am not sure if there would be a breaking change with exposing the tokens. I am just using the authAccount table to store it but I could just as easily manage another table myself.

**Comment by thomasballinger** [comment]
Let's ignore the breaking change thing with Convex Auth refreshing tokens at the same time that you are, Convex Auth isn't going to start managing refreshing auth tokens with oauth providers anytime soon.

So today you can write to users or another table in `afterUserCreatedOrUpdated` with this info. Is that good enough, or do you need the tokenSet to be added to another callback too?

**Comment by mez** [comment]
@thomasballinger are you saying the token info will be passed in the `afterUserCreatedOrUpdated` callback as an optional variable?

**Comment by mez** [comment]
Closing because it's not in the scope! I'll handle the process of access tokens in a different flow.
